### PR TITLE
MAINT: Fix issue in StataReader due to upstream changes

### LINKED ISSUE #35426: CI/COMPAT: read_stata failing in numpy dev pipeline
- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest version of pandas.

- [X] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Problem description

Looks like a change in numpy broke our `read_stata` code, so now we have a bunch of `read_stata` test errors popping up during CI in the numpy_dev pipeline. Maybe this [Numpy PR](https://github.com/numpy/numpy/pull/16943) somehow broke things? I don't think any other PR merged during the last 24 hours could have possibly done so, although I might be missing something.

Error example:
```
pandas/tests/io/test_stata.py:52: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/io/stata.py:1928: in read_stata
    data = reader.read()
pandas/io/stata.py:1646: in read
    cols_ = np.where(self.dtyplist)[0]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = ([<class 'numpy.int8'>, <class 'numpy.int16'>, <class 'numpy.int32'>, <class 'numpy.float32'>, <class 'numpy.float64'>, <class 'numpy.float32'>, ...],)
kwargs = {}
relevant_args = ([<class 'numpy.int8'>, <class 'numpy.int16'>, <class 'numpy.int32'>, <class 'numpy.float32'>, <class 'numpy.float64'>, <class 'numpy.float32'>, ...], None, None)

>   ???
E   ValueError: invalid __array_struct__
```

### PR DESCRIPTION
**Pull request by bashtage**
Avoid creating an array of dtypes to workaround NumPy future change

closes #35426

- [X] closes #35426
- [ ] tests added / passed
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by simonjayhawkins — APPROVED** [review]
Thanks @bashtage

**Comment by TomAugspurger** [comment]
Probably no reason to do this for 1.1.0. Are you OK with waiting for 1.1.1 @simonjayhawkins?

**Comment by simonjayhawkins** [comment]
> Probably no reason to do this for 1.1.0. Are you OK with waiting for 1.1.1 @simonjayhawkins?

no problem.

File: pandas/tests/window/test_ewm.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def test_ewma_halflife_without_times(halflife_with_times):
 @pytest.mark.parametrize("min_periods", [0, 2])
 def test_ewma_with_times_equal_spacing(halflife_with_times, times, min_periods):
     halflife = halflife_with_times
-    data = np.arange(10)
+    data = np.arange(10.)
```
Comment (simonjayhawkins) [inline pandas/tests/window/test_ewm.py:111]: IIUC you can revert this and just keep this PR scoped to the Stata change. see https://github.com/pandas-dev/pandas/issues/35426#issuecomment-665036331

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]

**Comment by simonjayhawkins** [comment]
moving milestone. The Stata tests are no longer failing with Linux py37_np_dev on master so this won't need backport.

**Comment by bashtage** [comment]
Should we just close this as unneeded?  The second fix does appear to be
needed or at least a good idea since assigning Nan to an int array has a
funny smell.

**Comment by simonjayhawkins** [comment]
> The second fix does appear to be needed or at least a good idea since assigning Nan to an int array has a funny smell.

Thanks. I'm not sure where we are with numpy-dev. We still have 20 fails on master. not sure whether further changes in numpy will resolve these or we still have work to do.

> Should we just close this as unneeded? 

up to you.

**Comment by bashtage** [comment]
@simonjayhawkins I reverted the second commit so this just a small, sensible always safe refactor.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @bashtage

**Comment by jbrockmendel** [comment]
Restarted the Travis build, LGTM pending green

**Comment by bashtage** [comment]
@jbrockmendel green.

**Comment by simonjayhawkins** [comment]
Thanks @bashtage

**Pull request merged**